exercises/list-ops/src/test/java/ListOpsTest.java

```diff
@@ -1,316 +1,186 @@
-        final List<Integer> actual = ListOps.concat(EMPTY_LIST, EMPTY_LIST);
-
-        assertNotNull(actual);
-        assertTrue(actual.isEmpty());
+    @Test
+    public void testTransformingNonEmptyList() {
+        assertEquals(
+                Arrays.asList(2, 4, 6, 8),
+                ListOps.map(Arrays.asList(1, 3, 5, 7), integer -> integer + 1 ));
     }
 
-    @Test
     @Ignore("Remove to run test")
-    public void shouldConcatenateOneEmptyAndOneNonEmptyLists() {
-        final List<Integer> list
-                = Collections.unmodifiableList(
-                        Arrays.asList(0, 1, 2, 3, 4)
-                );
-        final List<Integer> actual = ListOps.concat(list, EMPTY_LIST);
-        final List<Integer> expected
-                = Arrays.asList(0, 1, 2, 3, 4);
-
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertEquals(expected, actual);
+    @Test
+    public void testFoldLeftOnEmptyList() {
+        assertEquals(
+                new Double(2.0), // Boxing required for method overload disambiguation.
+                ListOps.foldLeft(
+                        Collections.<Double>emptyList(),
+                        2.0,
+                        (accumulation, current) -> current * accumulation));
     }
 
-    @Test
     @Ignore("Remove to run test")
-    public void shouldConcatenateOneNonEmptyAndOneEmptyLists() {
-        final List<Integer> list
-                = Collections.unmodifiableList(
-                        Arrays.asList(0, 1, 2, 3, 4)
-                );
-        final List<Integer> actual = ListOps.concat(EMPTY_LIST, list);
-        final List<Integer> expected
-                = Arrays.asList(0, 1, 2, 3, 4);
-
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertEquals(expected, actual);
+    @Test
+    public void testFoldLeftWithDirectionIndependentOperationOnNonEmptyList() {
+        assertEquals(
+                new Integer(15), // Boxing required for method overload disambiguation.
+                ListOps.foldLeft(
+                        Arrays.asList(1, 2, 3, 4),
+                        5,
+                        (accumulation, current) -> current + accumulation));
     }
 
-    @Test
     @Ignore("Remove to run test")
-    public void shouldConcatenateTwoListsWithSameElements() {
-        final List<Integer> list1 = Collections.unmodifiableList(
-                Arrays.asList(0, 1, 2, 3, 4)
-        );
-        final List<Integer> list2 = Collections.unmodifiableList(
-                Arrays.asList(1, 2, 3, 4, 5, 6)
-        );
-        final List<Integer> expected
-                = Arrays.asList(0, 1, 2, 3, 4, 1, 2, 3, 4, 5, 6);
-        final List<Integer> actual = ListOps.concat(list1, list2);
-
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertEquals(expected, actual);
+    @Test
+    public void testFoldLeftWithDirectionDependentOperationOnNonEmptyList() {
```

Comment:
How is this direction dependent? I understand that the operation in itself is but  with these numbers you would get the same answer no matter which direction you traverse the list wouldn't you? Wouldn't it be a better test if that wasn't the case?